compiler/src/parser.rs

```diff
@@ -170,11 +170,21 @@ peg::parser! {
               else { Constant::Int(n.parse().unwrap()) }
             }
 
+        rule escaped_char() -> String
+            = !['\\' | '\"'] c:$([_]) { String::from(c) }
+            / "\\" c:$(['t' | 'r' | 'n' | '\'' | '"' | '\\']) { String::from(c) }
+            / "\\u{" u:$(['a'..='f' | 'A'..='F' | '0'..='9']*<1,6>) "}" {
+                String::from(char::from_u32(u32::from_str_radix(u, 16).unwrap()).unwrap())
```

Comment:
I'm sure there's a nicer way of doing this, but this doesn't pull in any extra libraries and the 6 hex char limit means it can't overflow the u32